### PR TITLE
exp/disk_unlock: print hints if password mismatch

### PR DESCRIPTION
**Pull request by 10000TB**
If disk was locked with a different master password
it is likely users are unlocking wrong disks. Prompt
them to install dependency that rotates locking
schemes.

### CODE REVIEW

File: cmds/exp/disk_unlock/disk_unlock.go [hunk 1]
```diff
@@ -211,7 +211,7 @@ func main() {
 		// on the disk.
 		log.Fatalf("Security count expired on disk. Reset the password counter by power cycling the disk.")
 	case info.MasterPasswordRev != skmMPI:
-		log.Fatalf("Disk is locked with unknown master password ID: %X", info.MasterPasswordRev)
+		log.Fatalf("Disk is locked with unknown master password ID: %X (Do you have skm tools installed?)", info.MasterPasswordRev)
```
Comment (rminnich) [inline cmds/exp/disk_unlock/disk_unlock.go:214]: I don't know what this hint means :-) sorry

**Review by rminnich — APPROVED** [review]

File: cmds/exp/disk_unlock/disk_unlock.go (hunk 1 above)
Comment (10000TB) [reply cmds/exp/disk_unlock/disk_unlock.go:214]: ok, not a general good hint :(

skm tool, in our internal case, handles switching locking mechanism from one version to another. This go code hard codes a master password, which only works with v2.  I was hoping internal users seeing this error, not to panick, and know to install a dependency onto their machine(s) to fix the issue. Maybe I could be a bit more verbose in the log, but I feel that describing one scenario in the log can be mis-leading when users can run into other scenarios in real, which are both covered by the skm tools.

Comment (rminnich) [reply cmds/exp/disk_unlock/disk_unlock.go:214]: are these internal to google tools? Does this have meaning outside google?

Comment (orangecms) [reply cmds/exp/disk_unlock/disk_unlock.go:214]: https://timothyye.github.io/skm/

maybe add the URL?

Comment (10000TB) [reply cmds/exp/disk_unlock/disk_unlock.go:214]: google internal thing, I believe. But I believe if other places implement similar locking mechanism, this pkg can be useful, maybe that is why it is part of u-root, than in internal.

in our case, it is Storage Key Management, :) -- but skm can mean a lot of things

**Pull request merged**